### PR TITLE
Problem: no way to run the system with local pgsql

### PR DESCRIPTION
**Pull request by cognivore**
Solution: add makefile which orchestrates call of bootsttrap scripts. Improve secret granularity, including manual and autogenerated per-env configs!

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -34,6 +34,12 @@ defmodule ZhrDevs.MixProject do
       {:credo, "~> 1.7", only: [:dev, :test], runtime: false},
       {:dialyxir, "~> 1.2", only: [:dev], runtime: false},
 
+      # Phoenix just for phx.gen.secret :D
+      {:phoenix, "~> 1.7", only: [:dev], runtime: false},
+
+      # Ecto to control migrations.
+      {:ecto_sql, "~> 3.10"},
```
Comment (maslowalex) [inline mix.exs:41]: We will use `Ecto` really?

File: config/config.exs [hunk 2]
```diff
@@ -16,17 +16,6 @@ config :zhr_devs,
...
-  column_data_type: "jsonb",
-  serializer: Commanded.Serialization.JsonSerializer,
-  username: "postgres",
-  password: "postgres",
-  hostname: "localhost",
-  database: "zhr_devs_eventstore",
-  stacktrace: true,
-  show_sensitive_data_on_connection_error: true,
-  pool_size: 10
-
```
Comment (maslowalex) [inline config/config.exs:29]: I think that's the root of issue...

**Review by maslowalex — CHANGES_REQUESTED** [review]
Let's *either* get rid of `phoenix` end `ecto`, or migrate on that stack!

**Comment by cognivore** [comment]
Okay, I'll generate randomness from /dev/urandom, or with a one shot elixir script for portability!

Also sorry for deletion of the defaults! I didn't mean to.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# It is necessary, because we need to create roles in an insecure trust setting, before applying more secure settings.
+# The $(MAKE) command is a built-in variable in make that allows you to run make recursively.
+# See: https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html
+db: pgdata pgdata/sockets pgdata/postgresql.conf
+	$(MAKE) pg_ctl
+	priv/dev/create-role-phoenix.sh
+	$(MAKE) pgdata/pg_hba.conf
+	pg_ctl -D ./pgdata -l logfile restart
+	mix ecto.create || true
+	mix ecto.migrate || true
```
[the remark this answers is not in the document]
Comment (cognivore) [reply Makefile:32]: I'll keep it there for generality, but I'll remove it from the dependencies.

**Comment by cognivore** [comment]
Thank you for review, @maslowalex! Feel free to re-review the changes (most importantly, I returned the default parameters to the config.exs so that stuff doesn't break for you!).

**Pull request merged**